src/components/molecules/ProfilePictureInput/ProfilePictureInput.tsx

```diff
@@ -42,6 +44,16 @@ const ProfilePictureInput: React.FunctionComponent<PropsType> = ({
   const [error, setError] = useState("");
   const firebase = useFirebase();
   const uploadRef = useRef<HTMLInputElement>(null);
+  const queryParams = useQuery();
+  const venueId = queryParams.get("venueId");
+
+  const customAvatars = useAsync(async () => {
+    const storageRef = firebase.storage().ref();
+    const list = await storageRef.child(`/assets/avatars/${venueId}`).listAll();
+    const promises = list.items.map((item) => item.getDownloadURL());
+    const avatars: string[] = await Promise.all(promises);
+    return avatars;
+  }, []);
```

Comment:
Would this be better wrapped up in a custom hook of sorts? And/or in the `api/*` layer?

What storage security permissions are required to list all of the urls at that `storageRef` location? For this to work like this, it would basically need to be queryable by any user and/or potentially public?

Is `getDownloadURL` going to create/add a new key every time it's called, or will this always return the same URL?

Why wouldn't we just have an array of strings in the venue config itself or similar, rather than connecting directly to the storage like this?

Reply:
> Would this be better wrapped up in a custom hook of sorts? And/or in the api/* layer?

This is a very specific implementation and chances are it will never be reused because of this concept for default avatars.

> What storage security permissions are required to list all of the urls at that storageRef location? For this to work like this, it would basically need to be queryable by any user and/or potentially public?

This needs to be public because unregistered users will be fetching the default avatars.

> Is getDownloadURL going to create/add a new key every time it's called, or will this always return the same URL?

It seems to be returning the same URL every time.

> Why wouldn't we just have an array of strings in the venue config itself or similar, rather than connecting directly to the storage like this?

Even if we add this array of strings in the config, they'd still need to be uploaded in the storage or somewhere else. The only difference is that we wouldn't be fetching them directly. However, this means that in the admin part we'd have to upload them and then update the venue config. Where now, we only need to upload them.

Reply:
> This is a very specific implementation and chances are it will never be reused because of this concept for default avatars.

This was less about reuse and more about keeping heavier logic/etc outside of component files to aid in testing/etc.

> This needs to be public because unregistered users will be fetching the default avatars.

The storage collection itself doesn't need to be public/publicly listble/etc. Download URLs contain a 'baked in' key that makes their contents accessible. That's the only part of this that needs to be public I believe.

> It seems to be returning the same URL every time.

Cool

> Even if we add this array of strings in the config, they'd still need to be uploaded in the storage or somewhere else. The only difference is that we wouldn't be fetching them directly. However, this means that in the admin part we'd have to upload them and then update the venue config. Where now, we only need to upload them.

Makes sense. I don't see it being a huge issue that we would need to upload them then update venue config or similar. Though my stance on this is mostly going to come back to the answers on the 'publicly listable/accessible' stuff I mentioned above.